File: truss-compiler/src/main/java/truss/compiler/symbols/SymbolTreeWalker.java
package truss.compiler.symbols;

public class SymbolTreeWalker extends AbstractSymbolVisitor {
    @Override
    public void defaultVisit(Symbol symbol) {
        if (symbol instanceof ContainerSymbol) {
            visitList((ContainerSymbol)symbol);
        }
    }

    public void visitList(ContainerSymbol container) {
        for (Symbol symbol : container.getMembers()) {
            symbol.accept(this);
        }
    }
}

File: Truss.Compiler/Symbols/SymbolTreeWalker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Symbols {
    public class SymbolTreeWalker : AbstractSymbolVisitor {
        public override void DefaultVisit(Symbol symbol) {
            if (symbol is ContainerSymbol) {
                VisitList((ContainerSymbol)symbol);
            }
        }

        public virtual void VisitList(ContainerSymbol container) {
            foreach (var symbol in container.Members) {
                symbol.Accept(this);
            }
        }
    }
}
